src/node/export.ts

```diff
@@ -26,30 +33,77 @@ function selectFolder(callback: (result: string) => void) {
  * @param targetPath the path of the folder where the files will be copied.
  * @param callback a callback that will be called with the project name.
  */
-function collectFiles(
-  targetPath: string,
-  callback: (result: string) => Promise<void>,
-) {
-  csInterface.evalScript(`collectFiles("${targetPath}")`, callback);
+async function collectFiles(targetPath: string): Promise<string> {
+  const result = await evalScriptAsync(`collectFiles("${targetPath}")`);
+
+  const projectInfo: ProjectInfo = JSON.parse(result);
+
+  const projectName = path.basename(projectInfo.projectPath, '.aep');
+  const pathResolved = finalizePath(targetPath); // Normalize and resolve
+
+  await fsPromises.mkdir(path.join(pathResolved, projectName));
+  const projectDir = path.join(pathResolved, projectName);
+
+  const dest = path.join(projectDir, `${projectName}.aep`);
+  await fsPromises.copyFile(finalizePath(projectInfo.projectPath), dest);
+
+  await copyFonts(projectInfo.fonts, projectDir);
+  await copyFootage(projectInfo.footage, projectDir);
+
+  return projectDir;
 }
 
-/**
- * Replaces the tilde character at the start of a path with the user's home
- * directory.
- *
- * @example
- * untildify('~/Documents') // => '/Users/username/Documents'
- * @param {string} pathWithTilde - The path to transform.
- * @returns {string} The path with the tilde replaced.
- */
-function untildify(pathWithTilde: string): string {
-  if (typeof pathWithTilde !== 'string') {
-    throw new TypeError(`Expected a string, got ${typeof pathWithTilde}`);
+async function copyFonts(fonts: Fonts[], projectDir: string) {
+  if (fonts.length > 0) {
+    const fontsDir = path.join(projectDir, 'Fonts');
+    await fsPromises.mkdir(fontsDir);
+
+    const fontPromises = fonts.map(async (font) => {
+      const src = finalizePath(font.fontLocation);
+
+      const dest = path.join(
+        fontsDir,
+        `${font.fontName}.${font.fontExtension}`,
+      );
+      return fsPromises.copyFile(src, dest);
+    });
+
+    const errors = await runInParallelReturnRejected(fontPromises);
```

Comment:
Errors here will be errors thrown by `copyFile`. So, you will get the full message like `Error: ENOENT: no such file or directory, copyfile ...`. You can wrap `return fsPromises.copyFile(src, dest);` with try-catch, and `throw Error(src);` for example